include/ecs/component_pool.h

```diff
@@ -0,0 +1,87 @@
+#ifndef ECS_COMPONENT_POOL_H
+#define ECS_COMPONENT_POOL_H
+
+#include <vector>
+#include <unordered_map>
+
+#include "common.h"
+
+namespace arch::ecs {
+	class IComponentPool {
+	public:
+		virtual ~IComponentPool() = default;
+		virtual void on_entity_destroyed(EntityId entity_id) = 0;
+	};
+
+	template <typename T>
+	class ComponentPool : public IComponentPool {
+	public:
+		void insert(EntityId entity_id, T component);
+		void remove(EntityId entity_id);
+
+		T& get_data(EntityId entity_id);
+		void on_entity_destroyed(EntityId entity_id) override;
+
+	private:
+		std::vector<T> _component_array;

```

Comment:
Can't we just make _component_array std::unordered_map<EntityId, T> and not use these next maps at all?

Reply:
This vector was made here to ensure a tightly packed array of components to iterate over it, and modify components as fast as possible, it seems useless because we don't have something like an apply function implemented to showcase that, they just exist in a vectr